hazelcast/src/main/java/com/hazelcast/wan/WanEvent.java

```diff
@@ -16,29 +16,48 @@
 
 package com.hazelcast.wan;
 
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
 /**
  * Interface for all WAN replication messages
+ *
+ * @param <T> type of event data
  */
-public interface WanEvent {
+public interface WanEvent<T> {
     /**
      * Increments the count for the related event in the {@code counters}
      *
      * @param counters the WAN event counter
      */
-    void incrementEventCount(WanEventCounters counters);
+    void incrementEventCount(@Nonnull WanEventCounters counters);
```

Comment:
Mostly an `event` is a read-only object. Seems this method should belong to somewhere else. And in there a method `incrementEventCount(WanEvent wanEvent)` can be called?

Reply:
Maybe, yes, on the WanPublisher interface. But this requires a redesign and I would like to avoid doing it in this PR.

Reply:
Since the user doesn't need to invoke it and it's used internally, I'll move it into private API.

Reply:
Fixed.